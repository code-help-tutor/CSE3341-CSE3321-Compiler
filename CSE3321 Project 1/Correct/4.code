
program int X , Y , XY 	;

          input X, Y, XY;
		  
		  if not (X = Y)
		  then
			if X < Y
			then
				output 1;
			else
				output 2;
			;
		output 3;
		; 
		  
		input X, Y, XY;
		  
		  if not (X = Y)
		  then
			if X < Y
			then
				output 10;
			else
				output 20;
			;
		output 30;
		;
		
		input X, Y, XY;
		  
		  if  not X = Y)
		  then
			if X < Y
			then
				output 100;
			else
				output 200;
			;
		output 30;
		;

		input X, Y, XY;
		  
		if X = Y 
		then
			output 199;
		else
			if  not (X <= Y)
			then
				output 29;
			else
				output 39;
			;
		;
		
		
          end  
      
