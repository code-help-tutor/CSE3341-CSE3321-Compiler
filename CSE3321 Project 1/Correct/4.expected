ID[program]
ID[int]
ID[X]
COMMA
ID[Y]
COMMA
ID[XY]
SEMICOLON
ID[input]
ID[X]
COMMA
ID[Y]
COMMA
ID[XY]
SEMICOLON
IF
NOT
LPAREN
ID[X]
EQUAL
ID[Y]
RPAREN
THEN
IF
ID[X]
LESS
ID[Y]
THEN
ID[output]
CONST[1]
SEMICOLON
ELSE
ID[output]
CONST[2]
SEMICOLON
SEMICOLON
ID[output]
CONST[3]
SEMICOLON
SEMICOLON
ID[input]
ID[X]
COMMA
ID[Y]
COMMA
ID[XY]
SEMICOLON
IF
NOT
LPAREN
ID[X]
EQUAL
ID[Y]
RPAREN
THEN
IF
ID[X]
LESS
ID[Y]
THEN
ID[output]
CONST[10]
SEMICOLON
ELSE
ID[output]
CONST[20]
SEMICOLON
SEMICOLON
ID[output]
CONST[30]
SEMICOLON
SEMICOLON
ID[input]
ID[X]
COMMA
ID[Y]
COMMA
ID[XY]
SEMICOLON
IF
NOT
ID[X]
EQUAL
ID[Y]
RPAREN
THEN
IF
ID[X]
LESS
ID[Y]
THEN
ID[output]
CONST[100]
SEMICOLON
ELSE
ID[output]
CONST[200]
SEMICOLON
SEMICOLON
ID[output]
CONST[30]
SEMICOLON
SEMICOLON
ID[input]
ID[X]
COMMA
ID[Y]
COMMA
ID[XY]
SEMICOLON
IF
ID[X]
EQUAL
ID[Y]
THEN
ID[output]
CONST[199]
SEMICOLON
ELSE
IF
NOT
LPAREN
ID[X]
LESS
EQUAL
ID[Y]
RPAREN
THEN
ID[output]
CONST[29]
SEMICOLON
ELSE
ID[output]
CONST[39]
SEMICOLON
SEMICOLON
SEMICOLON
END
